### PR TITLE
[WIP] Update to PF4 v4

### PR DESCRIPTION
**Pull request by lgalis**
Screenshot from 2020-06-16 17-39-45
Screenshot from 2020-06-18 09-24-39
Screenshot from 2020-06-16 17-40-30
Screenshot from 2020-06-16 17-40-50
Screenshot from 2020-06-19 12-32-13

### CODE REVIEW

**Comment by sbuenafe-rh** [comment]
@lgalis we want the new style for the tabs. thanks for checking!

**Comment by Hyperkid123** [comment]
@lgalis can you check the tests?

**Comment by lgalis** [comment]
@Hyperkid123  - the Create and Remove workflow buttons are actually not visible in the UI, even with the beta.2 version of the front end components - will fix those then the tests as needed.

**Comment by Hyperkid123** [comment]
@lgalis ok. If It its a bug on the FCE side, let me know and I will fix it ASAP.

**Comment by lgalis** [comment]
@Hyperkid123  - it does seem to be an issue with the PrimaryToolbar action buttons. As you can see - no buttons are visible in the toolbar in the workflows page. On a mobile -size screen, the kebab and buttons are both appear: 
Screenshot from 2020-06-17 10-47-07

**Comment by Hyperkid123** [comment]
Ah that issue was fixed already and there should be 2.1.0-beta.1 version available

**Comment by lgalis** [comment]
@Hyperkid123 - before posting my comment about the primary toolbar, I actually tried both beta.1 and beta.2 ( I had seen the very similar github issue that was fixed in beta.1) - the result is the same, no buttons.

**Comment by Hyperkid123** [comment]
@lgalis you are right there is another CSS bug that hides the first action item for some reason.  It should be fixed today.

**Comment by Hyperkid123** [comment]
@lgalis fix should be published in version `2.1.0-beta.6`

**Comment by lgalis** [comment]
> @lgalis fix should be published in version `2.1.0-beta.6`

@Hyperkid123 - thank you.

File: package.json [hunk 1]
```diff
@@ -3,15 +3,15 @@
   "version": "0.1.0",
   "private": false,
   "dependencies": {
-    "@data-driven-forms/pf4-component-mapper": "2.4.8",
-    "@data-driven-forms/react-form-renderer": "2.4.8",
-    "@patternfly/react-core": "^3.158.3",
-    "@patternfly/react-table": "^2.28.47",
-    "@patternfly/react-icons": "^3.15.17",
+    "@data-driven-forms/form-builder": "0.0.10-rc7",
```
Comment (Hyperkid123) [inline package.json:6]: Form builder is not used by approval

File: package.json [hunk 2]
```diff
@@ -22,14 +22,16 @@
...
-    "react-select": "^3.0.4",
-    "redux": "^3.7.2",
+    "redux": "^4.0.4",
     "redux-logger": "^3.0.6",
-    "redux-mock-store": "^1.5.3",
     "redux-promise-middleware": "^5.1.1",
-    "redux-thunk": "^2.3.0"
+    "redux-thunk": "^2.3.0",
+    "smoothscroll-polyfill": "^0.4.4",
+    "styled-components": "^5.0.1",
```
Comment (Hyperkid123) [inline package.json:32]: ^^

File: package.json [hunk 3]
```diff
@@ -22,14 +22,16 @@
...
+    "whatwg-fetch": "3.0.0",
```
Comment (Hyperkid123) [inline package.json:33]: ^^

File: package.json [hunk 4]
```diff
@@ -22,14 +22,16 @@
...
     "react-router-dom": "^5.1.2",
-    "react-select": "^3.0.4",
-    "redux": "^3.7.2",
+    "redux": "^4.0.4",
     "redux-logger": "^3.0.6",
-    "redux-mock-store": "^1.5.3",
     "redux-promise-middleware": "^5.1.1",
-    "redux-thunk": "^2.3.0"
+    "redux-thunk": "^2.3.0",
+    "smoothscroll-polyfill": "^0.4.4",
```
Comment (Hyperkid123) [inline package.json:31]: ^^

File: src/forms/add-workflow.schema.js [hunk 5]
```diff
@@ -19,7 +19,7 @@ const addWorkflowSchema = (intl) => ({
     fields: [{
       name: 'general-information',
       showTitle: true,
-      customTitle: <Title size="md"> { <FormattedMessage id="enter-info" defaultMessage="Enter your information" /> } </Title>,
+      customTitle: <Title headingLevel="h3" size="md"> { <FormattedMessage id="enter-info" defaultMessage="Enter your information" /> } </Title>,
```
Comment (Hyperkid123) [inline src/forms/add-workflow.schema.js:22]: The default value for Title used to be `h1` I am guessing the sizes did not match?

File: src/presentational-components/shared/table-toolbar-view.js [hunk 6]
```diff
@@ -132,6 +132,7 @@ export const TableToolbarView = ({
             className="table-fix"
             sortBy={ sortBy }
             onSort={ onSort }
+            canSelectAll={ true }
```
Comment (Hyperkid123) [inline src/presentational-components/shared/table-toolbar-view.js:135]: `canSelectAll={ true } => canSelectAll` is preferable

File: src/smart-components/error-pages/common-api-error.js [hunk 7]
```diff
@@ -35,7 +35,7 @@ const CommonApiError = () => {
             <EmptyStateIcon icon={ LockIcon } />
           </div>
           <div>
-            <Title size="lg">{ TITLES[pathname] }</Title>
+            <Title headingLevel="h5" size="lg">{ TITLES[pathname] }</Title>
```
Comment (Hyperkid123) [inline src/smart-components/error-pages/common-api-error.js:38]: Same here with the default `headingLevel`

File: package.json [hunk 8]
```diff
@@ -3,15 +3,14 @@
   "version": "0.1.0",
   "private": false,
   "dependencies": {
-    "@data-driven-forms/pf4-component-mapper": "2.4.8",
-    "@data-driven-forms/react-form-renderer": "2.4.8",
-    "@patternfly/react-core": "^3.158.3",
-    "@patternfly/react-table": "^2.28.47",
-    "@patternfly/react-icons": "^3.15.17",
+    "@data-driven-forms/pf4-component-mapper": "^2.5.0-beta.1",
```
Comment (Hyperkid123) [inline package.json:6]: there is a stable version of DDF `2.5.3` for both the mapper and renderer. Sorry I did not notice earlier.

File: package.json [hunk 9]
```diff
@@ -76,14 +75,15 @@
...
-    "git-revision-webpack-plugin": "^3.0.3",
+    "file-loader": "^4.3.0",
+    "git-revision-webpack-plugin": "^3.0.6",
     "glob": "^7.1.6",
     "html-replace-webpack-plugin": "^2.5.3",
     "html-webpack-plugin": "^3.0.4",
     "identity-obj-proxy": "^3.0.0",
     "jest": "^26.0.1",
     "jest-cli": "^26.0.1",
+    "koa-connect": "^2.0.1",
```
Comment (Hyperkid123) [inline package.json:86]: This does not seem to be used.

**Comment by Hyperkid123** [comment]
OK, I think it's ready. Let's wait until we can merge

**Comment by Hyperkid123** [comment]
closing in favor of #308. We needed quickly to update the FCE dependencies. It is just a rebase of this PR with additional commit.

**Pull request closed without merging**